aspnetcore/security/authentication/certauth.md

```diff
@@ -0,0 +1,243 @@
+        .AddCertificate(options =>
+        {
+            options.Events = new CertificateAuthenticationEvents
+            {
+                OnCertificateValidated = context =>
+                {
+                    var validationService =
+                        context.HttpContext.RequestServices.GetService<ICertificateValidationService>();
+                    
+                    if (validationService.ValidateCertificate(context.ClientCertificate))
+                    {
+                        var claims = new[]
+                        {
+                            new Claim(ClaimTypes.NameIdentifier, context.ClientCertificate.Subject, ClaimValueTypes.String, context.Options.ClaimsIssuer),
+                            new Claim(ClaimTypes.Name, context.ClientCertificate.Subject, ClaimValueTypes.String, context.Options.ClaimsIssuer)
+                        };
+
+                        context.Principal = new ClaimsPrincipal(new ClaimsIdentity(claims, context.Scheme.Name));
+                        context.Success();
+                    }                     
+
+                    return Task.CompletedTask;
+                }
+            };
+        });
+```
+Note that conceptually the validation of the certification is an authorization concern, and putting a check on, for example, an issuer or thumbprint in an authorization policy rather 
+than inside OnCertificateValidated() is perfectly acceptable.
+
+## <a name="hostConfiguration"></a>Configuring your host to require certificates
+
+### Kestrel
+
+In program.cs configure `UseKestrel()` as follows.
+
+```c#
+public static IWebHost BuildWebHost(string[] args)
+    => WebHost.CreateDefaultBuilder(args)
+    .UseStartup<Startup>()
+    .ConfigureKestrel(options =>
+    {
+        options.ConfigureHttpsDefaults(opt =>
+        {
+            opt.ClientCertificateMode = ClientCertificateMode.RequireCertificate;
+        });
+    })
+    .Build();
+```
+You must set the `ClientCertificateValidation` delegate to `CertificateValidator.DisableChannelValidation` in order to stop Kestrel using the default OS certificate validation routine and, 
+instead, letting the authentication handler perform the validation.
+
+### IIS
+
+In the IIS Manager 
+
+1. Select your Site in the Connections tab.
+2. Double click the SSL Settings in the Features View window.
+3. Check the `Require SSL` Check Box and select the `Require` radio button under Client Certificates.
+
+![Client Certificate Settings in IIS](README-IISConfig.png "Client Certificate Settings in IIS")
+
+### Azure
+
+See the [Azure documentation](https://docs.microsoft.com/en-us/azure/app-service/app-service-web-configure-tls-mutual-auth) 
+to configure Azure Web Apps then add the following to your application startup method, `Configure(IApplicationBuilder app)` add the 
+following line before the call to `app.UseAuthentication();`
+
+```c#
+app.UseCertificateHeaderForwarding();
```

Comment:
Options will be needed for this. Update https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-2.2 to cover this and link to it rather than covering here.

Reply:
Added the link and updated the other doc